### ISSUE TITLE
Feature Request: Detect "Article + Param + Filler Phrase" Redundant Descriptions

### ISSUE THREAD

**Issue by nijikon** [mensfeld/yard-lint#32]
## Summary

Add a new pattern `ArticleParamPhrase` to the `Tags/RedundantParamDescription` rule that detects descriptions which restate the parameter name with filler words like "being performed" or "being acted upon".

## Problem

Currently, the `ArticleParam` pattern only catches exact 2-word matches like:

```ruby
# @param appointment [Appointment] The appointment  # <- CAUGHT
```

But semantically equivalent redundant descriptions slip through:

```ruby
# @param action [Symbol] The action being performed.  # <- NOT CAUGHT
# @param appointment [Appointment] The appointment being acted upon  # <- NOT CAUGHT
```

These descriptions add no meaningful information beyond what the parameter name and type already convey.

## Expected Behavior

The following should be flagged as redundant:

```ruby
# @param action [Symbol] The action being performed.
# @param appointment [Appointment] The appointment being acted upon
# @param user [User] The user to process
# @param callback [Proc] A callback that is invoked
# @param data [Hash] The data being passed
```

Suggested fix for all of the above:

```ruby
# @param action [Symbol]
# @param appointment [Appointment]
# @param user [User]
# @param callback [Proc]
# @param data [Hash]
```

Or, if a description is needed, provide meaningful information:

```ruby
# @param action [Symbol]
#   Valid values: :edit, :update, :calendar_move, :delete, :destroy
# @param appointment [Appointment]
# @param identified_changes [Array<Symbol>] Array of change symbols detected during edit
```

## What Should NOT Be Flagged

Descriptions that start similarly but continue with meaningful content:

```ruby
# @param action [Symbol] The action that determines which confirmation options to display
# @param appointment [Appointment] The appointment whose recurring series will be modified
# @param user [User] The user who must have admin privileges to proceed
```

These are valid because they provide context beyond the parameter name.

### Pattern: `ArticleParamPhrase`

Detects descriptions matching:

```
<Article> <param_name> <connector> [<low_value_verb>...]
```

Where:
- **Article**: The, the, A, a, An, an (configurable via existing `Articles` setting)
- **param_name**: Exact match with the parameter name (case-insensitive)
- **connector**: being, to, that, which, for (new `LowValueConnectors` setting)
- **low_value_verb**: Stems like perform, process, use, handle, act, etc. (new `LowValueVerbStems` setting)

### Detection Logic

```ruby
# Pseudocode
if word_count >= 3 && word_count <= 5
  if words[0] matches article &&
     words[1].downcase == param_name.downcase &&
     words[2] matches connector

    remaining_words = words[3..]
    if remaining_words.empty? || all_are_low_value_verbs?(remaining_words)
      flag as 'article_param_phrase'
    end
  end
end
```

### Simple Stemming (No Dependencies)

To match verb forms without adding external dependencies, use inline suffix stripping:

```ruby
word.downcase.gsub(/ed$|ing$|s$|ly$/, '')
```

This transforms:
- `performed` → `perform`
- `processing` → `process`
- `invoked` → `invok` (matches `invoke` stem)
- `acted` → `act`

### Configuration Additions

```yaml
Tags/RedundantParamDescription:
  # Existing settings...

  # NEW: Connectors that link param name to low-value verbs
  LowValueConnectors:
    - being
    - to
    - that
    - which
    - for

  # NEW: Verb stems indicating low-value descriptions
  LowValueVerbStems:
    - perform
    - process
    - use
    - handle
    - act
    - apply
    - pass
    - provide
    - give
    - specify
    - invoke
    - call
    - execute
    - operate
    - work
    - do
    - make
    - take

  EnabledPatterns:
    # Existing patterns...
    ArticleParamPhrase: true  # NEW
```

### Message

```
The @param description 'The action being performed' just rephrases the parameter name
with filler words. Consider removing it (`@param action [Type]`) or adding specific
details like valid values, constraints, or purpose.
```

### Flagged (Redundant)

| Description | Pattern Match |
|-------------|---------------|
| `The action being performed` | Article + param + "being" + generic verb |
| `The appointment being acted upon` | Article + param + "being" + generic verb + "upon" |
| `A callback to invoke` | Article + param + "to" + generic verb |
| `The user that is processed` | Article + param + "that" + filler + generic verb |
| `The data being passed` | Article + param + "being" + generic verb |

### Not Flagged (Valid)

| Description | Why Valid |
|-------------|-----------|
| `The action that determines which options to show` | >5 words, has meaningful content |
| `Valid values: :edit, :update` | Doesn't start with Article + param |
| `The action` | Caught by existing `ArticleParam` pattern |
| `Must be a valid Symbol from ALLOWED_ACTIONS` | Doesn't match pattern structure |

## Research Background

This enhancement is informed by recent research on detecting redundant/obvious comments:

- [Deep Learning to Detect Redundant Method Comments (CRAIC)](https://arxiv.org/abs/1806.04616) - Defines "comment entailment from code" where high entailment indicates redundancy
- [Towards Automated Detection of Inline Code Comment Smells](https://arxiv.org/html/2504.18956v1) - Taxonomy includes "obvious comments" that restate what code does
- [FSE 2024: Detecting Code Comment Inconsistencies](https://dl.acm.org/doi/10.1145/3663529.3664458) - LLM-based approaches show code context improves detection

For a linting tool, heuristic pattern matching provides the best balance of:
- **Precision**: Low false positive rate with conservative patterns
- **Speed**: No external API calls or model inference
- **Configurability**: Users can tune word lists for their domain
- **No dependencies**: Pure Ruby implementation

### Token Overlap Score

Measure what percentage of the description is just the parameter name with stopwords removed. Rejected because:
- Higher false positive rate
- More complex to explain to users
- Harder to configure

### Entropy-Based Detection

Count "information-bearing" words after removing stopwords and generic terms. Could be added as a separate opt-in pattern (`LowEntropy`) in the future for users who want more aggressive detection.
